Enhance Authentication Flow and JWT Handling

Wejdene-CH opened the pull request:
The changes aim to enhance the overall security and performance of the authentication system.

Wejdene-CH committed:
FNZ-TimeTracking-2 :
-Modified authController class to improve authentication flow( just i added some comments)
- Enhanced authService class for better token handling
- Updated JwtUtil class for optimized JWT operations
- Adjusted User class to align with authentication changes
- Refined userRepository class for improved data handling

Commit:
Merge branch 'main' into feature_fnz-TimeTracking_2

# Conflicts:
#	src/main/java/com/fnz/TimeTracking/controller/AuthController.java
#	src/main/java/com/fnz/TimeTracking/service/AuthService.java
#	src/main/java/com/fnz/TimeTracking/service/JwtUtil.java

Commit:
Merge branch 'main' into feature_fnz-TimeTracking_2

# Conflicts:
#	src/main/java/com/fnz/TimeTracking/controller/AuthController.java
#	src/main/java/com/fnz/TimeTracking/service/AuthService.java
#	src/main/java/com/fnz/TimeTracking/service/JwtUtil.java

baccartaoufik commented on src/main/java/com/fnz/TimeTracking/model/Utilisateur.java:
```diff
@@ -23,10 +23,10 @@ public class Utilisateur {
-    //@Getter
-    //private String password; // Add a password field
+    @Getter
+    private String username; // Add a username field
```
Why adding a username and a password ?

We are using a facial recongnition auth system.

a user will be identified by the photo field, which will be a face print

baccartaoufik reviewed (changes requested):
Clean up code, remove comments, extra spaces.
Revert change in user model.
Explain what are claims in JWT token.

Wejdene-CH committed:
FNZ-TimeTracking-2 :making changes
JWT claims are the core information that JWTs transmit (kinda like the letter inside a sealed envelope).

The JWT claims included in the payload determine which information the JWT communicates (i.e., user identity, permissions, expiration of JWT, to name a few).
In our case, when using JWTs for authentication based on face recognition:
-Subject (sub): This could represent the unique identifier of the user authenticated via face recognition, such as a user ID.
-Expiration Time (exp): Specifies the time when the JWT will expire and should no longer be accepted for authentication.

Commit:
FNZ-TimeTracking-2 :Adding missing deps
JWT claims are the core information that JWTs transmit (kinda like the letter inside a sealed envelope).

Commit:
FNZ-TimeTracking-2 :disable default spring security
JWT claims are the core information that JWTs transmit (kinda like the letter inside a sealed envelope).
